framework/src/modules/interoperability/mainchain/store.ts

```diff
@@ -39,6 +45,31 @@ export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
 		console.log(ccu, ccm);
 	}
 
+	public async bounce(ccm: CCMsg): Promise<void> {
+		const terminatedStateSubstore = this.getStore(
+			MODULE_ID_INTEROPERABILITY,
+			STORE_PREFIX_TERMINATED_STATE,
+		);
+
+		const terminatedStateAccountExists = await terminatedStateSubstore.has(
+			getIDAsKeyForStore(ccm.sendingChainID),
+		);
```

Comment:
we can also bundle them to another function within baseStore class `hasTerminatedStateAccount(chainID)` so that we can reuse this everytime its used. We can do it for get/set/has moving forward so that we don't have to find the right constants to get store and write repetitive code.